cfme/tests/ansible/test_embedded_ansible_basic.py

```diff
@@ -131,6 +133,35 @@ def test_embedded_ansible_repository_crud(ansible_repository, wait_for_ansible):
     assert view.description.value == updated_description
 
 
+@pytest.mark.rhel_testing
+@pytest.mark.tier(1)
+def test_embedded_ansible_repository_branch_crud(appliance, wait_for_ansible):
+    """
+    Ability to add repo with branch (without SCM credentials).
+
+    Polarion:
+        assignee: sbulage
+        casecomponent: Ansible
+        caseimportance: critical
+        initialEstimate: 1/12h
+        tags: ansible_embed
+    """
+    repositories = appliance.collections.ansible_repositories
+    repository = repositories.create(
+        name=fauxfactory.gen_alpha(),
+        url="https://github.com/ManageIQ/integration_tests_playbooks.git",
+        description=fauxfactory.gen_alpha(),
+        scm_branch="second_playbook_branch",
+    )
+    view = navigate_to(repository, "Details")
+    assert (
+        view.entities.summary("Repository Options").get_text_of("SCM Branch")
+        == repository.scm_branch
+    )
+
+    repository.delete()
```

Comment:
I think we need to check `.exists` after creation and `not .exists` after deletion of a repository as test pointing to crude.

Reply:
Above `assert` statement and `view` covers the `.exists` of repository.

Reply:
it doesn't fully cover this case. if your test fails on assert, your repository won't be deleted.

Reply:
I will add `repository.delete_if_exists()` in finalizer.